src/main/java/seedu/address/ui/TaskCard.java

```diff
@@ -50,9 +51,9 @@ public TaskCard(Task task, int displayedIndex) {
         id.setText(displayedIndex + ". ");
         description.setText(task.getDescription().value);
         taskId.setText("Task ID: #" + task.getId());
-        assigned.setText("Assigned: " + task.getDriver().get().getSchedule() + " @ "
-            + task.getDriver().get().getName().fullName + " (#" + task.getDriver().get().getId()
-            + ") - " + task.getDriver().get().getPhone());
+        assigned.setText("Assigned: " + task.getEventTime().map(EventTime::toString).orElse("NOT ALLOCATED") + " @ "
+                + task.getDriver().get().getName().fullName + " (#" + task.getDriver().get().getId()
+                + ") - " + task.getDriver().get().getPhone());
```

Comment:
i think using map here is redundant cos it will automatically change to a string when u concatenate.

so task.getEventTime().orElse("NOT ALLOCATED") will do

Reply:
Nah, we are working with optionals here. I tried but couldn't compile, because `task.getEventTime()` returns an `Optional<EventTime>`. Then the type of the element inside`orElse` has to be `EventTime`